public/components/event_analytics/utils/utils.tsx

```diff
@@ -459,3 +462,57 @@ export const getContentTabTitle = (tabID: string, tabTitle: string) => {
     </>
   );
 };
+
+/**
+ * Used to fill in missing empty data where x is an array of time values and there are only x
+ * values when y is non-zero.
+ * @param xVals all x values being used
+ * @param yVals all y values being used
+ * @param selectedInterval Moment unitOfTime used to dictate how long each interval is
+ * @param startTime starting time of x values
+ * @param endTime ending time of x values
+ * @returns an object with buckets and values where the buckets are all of the new x values and
+ * values are the corresponding values which include y values that are 0 for empty data
+ */
+export const fillTimeDataWithEmpty = (
+  xVals: string[],
+  yVals: number[],
+  selectedInterval: any,
+  startTime: string,
+  endTime: string
+) => {
```

Comment:
Do we also want to add return type for this function?

Reply:
I've added the return type